src/components/StyledMarkdown.js

```diff
@@ -0,0 +1,100 @@
+import React from 'react'
+import {
+  Dimensions,
+  Linking,
+  Platform,
+  WebView,
+} from 'react-native'
+import EStyleSheet from 'react-native-extended-stylesheet'
+import defaultHTML from '../utils/default-md-html'
+
+import { removeMDTab } from '../utils/remove-md-tab'
+
+const MarkdownIt = require('markdown-it'),
+    md = new MarkdownIt({ linkify: true, breaks: true }).use(removeMDTab)
+
+const WEBVIEW_REF = 'WEBVIEW_REF'
+
+class StyledMarkdown extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = { height: 0 }
+  }
+
+  onShouldStartLoadWithRequest = (event) => {
+    if (event.url.indexOf('http') >= 0 ) {
+      this.refs[WEBVIEW_REF].stopLoading()
+      Linking.openURL(event.url)
+      return false
+    } else {
+      return true
+    }
+  }
+
+  onNavigationStateChange = (navState) => {
+    if (Platform.OS === 'android') {
+      this.onShouldStartLoadWithRequest(navState)
+    }
+    if (Number(navState.title) > 0) {
+      const htmlHeight = Number(navState.title) //convert to number
+      this.setState({height: htmlHeight})
+      this.props.onReceivedHeight(htmlHeight)
+    }
+  }
+
+  render() {
+    //Markdownz uses markdown-it-imsize for image sizes, however this package
+    //requires fs, which is a node core module and not available natively
+    //Also, we can't use the image sizes.  This regexp removes them, prior to
+    //being sent to renderer, or else they're interpreted as links
+    let markdown = this.props.markdown.replace(/ =[0-9]+x[0-9]*\)/g, '\)')
+    let result = md.render(markdown)
+
+    const resultHTML = defaultHTML
+      .replace('$body', result)
+      .replace('$extraCSS', this.props.extraCSS)
+
+    const displayWidth = this.props.width || Dimensions.get('window').width - 40
+
+    const webviewComponent =
+      <WebView
+        ref={WEBVIEW_REF}
```

Comment:
React changed the way they handle `refs`. Instead of using string-based refs, callbacks are now preferred. There are some good docs from [React Native](https://facebook.github.io/react-native/docs/direct-manipulation.html) and [React](https://facebook.github.io/react/docs/refs-and-the-dom.html).